synfig-studio/src/gui/workarea.cpp

```diff
@@ -1468,15 +1484,17 @@ WorkArea::on_drawing_area_event(GdkEvent *event)
 	        break;
 		}
 		case DRAG_GUIDE: {
-			if (!rotate_guide){
-				curr_guide->point[0] = mouse_pos[0];
-				curr_guide->point[1] = mouse_pos[1];
-			} else if(rotate_guide && !from_ruler_event) {
+			if(!get_lock_guides()) {
+				if (!rotate_guide) {
+					curr_guide->point[0] = mouse_pos[0];
+					curr_guide->point[1] = mouse_pos[1];
+				} else if(rotate_guide && !from_ruler_event) {
 					float slope = (mouse_pos[1] - curr_guide->point[1])/(mouse_pos[0] - curr_guide->point[0]);
 					//just change the angle of the ruler
 					curr_guide->angle = synfig::Angle::rad(atan(slope));
+				}
+				drawing_area->queue_draw();
 			}
-			drawing_area->queue_draw();
 	        break;
```

Comment:
If `DRAG_GUIDE` is never set when `get_lock_guides()` is true, so you don't need to add this `if` here.

However, you did miss the case when user tries to drag a guide.
https://github.com/synfig/synfig/blob/768188305412e231309cddf467c58bcc304863d9/synfig-studio/src/gui/workarea.cpp#L1307-L1316

I searched for when drag mode is set to `DRAG_GUIDE` to find it.